packages/2d/src/components/Line.ts

```diff
@@ -265,15 +265,25 @@ export class Line extends Shape {
     context.beginPath();
     if (this.endArrow() instanceof Node) {
       const endArrow = this.endArrow() as Node;
-      endArrow.position(endPoint);
+      context.translate(endPoint.x, endPoint.y);
+      context.rotate(Math.atan2(endTangent.y, endTangent.x));
+
       endArrow.render(context);
+
+      context.rotate(-Math.atan2(endTangent.y, endTangent.x));
```

Comment:
We should use `save/restore` instead. Reverting the transform like that may be too imprecise due to floating-point errors.